### PR TITLE
Backport URP FSR to 2021.2

### PR DESCRIPTION
**Pull request by gmitrano-unity**
### Purpose of this PR
Backport URP FSR to 2021.2

This change backports the FidelityFX Super Resolution feature in URP to the 2021.2 branch to allow users to take advantage of it earlier.

---
### Testing status
Manual local testing on Windows & DX11

---
### Comments to reviewers
This change needs a full round of QA testing on mobile

Important Note: This change will **NOT** function correctly on iOS/Metal platforms until a particular shader compiler fix is backported to 2021.2 as well! (https://fogbugz.unity3d.com/f/cases/1387697/) Without this fix, users will experience a black screen when FSR is enabled.

### CODE REVIEW

**Review by phi-lira — APPROVED** [review]
LGTM. @gmitrano-unity it's a big PR, if you could highlight the differences from master I would appreciate.

File: com.unity.render-pipelines.universal/Runtime/Passes/PostProcessPass.cs [hunk 1]
```diff
@@ -1395,6 +1413,132 @@ void RenderFinalPass(CommandBuffer cmd, ref RenderingData renderingData)
...
+                    {
+                        m_Materials.scalingSetup.EnableKeyword(ShaderKeywordStrings.Fxaa);
+                    }
+
+                    if (doLateFsrColorConversion)
+                    {
+                        m_Materials.scalingSetup.EnableKeyword(ShaderKeywordStrings.Gamma20);
+                    }
+
+                    cmd.GetTemporaryRT(ShaderConstants._ScalingSetupTexture, tempRtDesc, FilterMode.Point);
```
Comment (gmitrano-unity) [inline com.unity.render-pipelines.universal/Runtime/Passes/PostProcessPass.cs:1454]: ^ This one

File: com.unity.render-pipelines.universal/Runtime/Passes/PostProcessPass.cs [hunk 2]
```diff
@@ -1395,6 +1413,132 @@ void RenderFinalPass(CommandBuffer cmd, ref RenderingData renderingData)
...
+                            case ImageUpscalingFilter.FSR:
+                            {
+                                m_Materials.easu.shaderKeywords = null;
+
+                                var upscaleRtDesc = tempRtDesc;
+                                upscaleRtDesc.width = cameraData.pixelWidth;
+                                upscaleRtDesc.height = cameraData.pixelHeight;
+
+                                // EASU
+                                cmd.GetTemporaryRT(ShaderConstants._UpscaledTexture, upscaleRtDesc, FilterMode.Point);
```
Comment (gmitrano-unity) [inline com.unity.render-pipelines.universal/Runtime/Passes/PostProcessPass.cs:1495]: ^ This one too

**Review by gmitrano-unity — COMMENTED** [review]
The main difference between this PR and the one in master is that this one uses temporary RTs instead of RTHandles for the intermediate render targets in the post processing pass.

**Review by ernestasKupciunas — APPROVED** [review]
Looks good.
Tested on Android, iOS, Apple TV and OSX paltforms. I have found a few issues - but these are the same as in the original PR. I will report discovered issues later.
Testing doc: https://docs.google.com/document/d/1IX5-phePpDUbwV9ry0hzKqpP1I6ekmA31IFthrdMJkA/edit

**Review by phi-lira — APPROVED** [review]

File: com.unity.render-pipelines.universal/Runtime/Passes/PostProcessPass.cs (hunk 2 above)
Comment (phi-lira) [reply com.unity.render-pipelines.universal/Runtime/Passes/PostProcessPass.cs:1495]: Sounds good

**Comment by phi-lira** [comment]
[Yamato] URP_Foundation on Win_DX11_playmode_XR_mono_Linear on version 2021.2 - test is failing on 2021.2/staging branch as well.

**Pull request merged**